src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueHealthFactorLogic.sol

```diff
@@ -0,0 +1,122 @@
+
+        if (collateralAmountInBorrowUnits == 0) return 0;
+
+        // calculate the currentPositionLTV then compare it against the max lltv for this position
+        uint256 LTV_PRECISION = 1e18;
+        uint256 currentPositionLTV = borrowAmount.mulDivUp(LTV_PRECISION, collateralAmountInBorrowUnits);
+
+        console.log(
+            "positionMaxLTV: %s, currentPositionLTV: %s, collateralAmountInBorrowUnits: %s",
+            _market.lltv,
+            currentPositionLTV,
+            collateralAmountInBorrowUnits
+        );
+        // convert LTVs to HF
+        currentHF = (_market.lltv).mulDivDown(1e18, currentPositionLTV);
+    }
+
+    /**
+     * @dev helper function that returns actual supply position amount for caller according to MB market accounting. This is alternative to using the MB periphery libraries that simulate accrued interest balances.
+     * NOTE: make sure to call `accrueInterest()` on respective market before calling these helpers
+     */
+    function _userSupplyBalance(Id _id, address _user) internal view returns (uint256) {
+        Market memory market = morphoBlue.market(_id);
+        // this currently doesn't account for interest, that needs to be done before calling this helper.
+        return (morphoBlue.supplyShares(_id, _user).toAssetsDown(market.totalSupplyAssets, market.totalSupplyShares));
+    }
+
+    /**
+     * @dev helper function that returns actual collateral position amount for caller according to MB market accounting. This is alternative to using the MB periphery libraries that simulate accrued interest balances.
+     * NOTE: make sure to call `accrueInterest()` on respective market before calling these helpers
+     */
+    function _userCollateralBalance(Id _id, address _user) internal view virtual returns (uint256) {
+        // return uint256(morphoBlue.position(_id, address(this)).collateral);
+        return morphoBlue.collateral(_id, _user);
+    }
+
+    /**
+     * @dev helper function that returns actual borrow position amount for caller according to MB market accounting. This is alternative to using the MB periphery libraries that simulate accrued interest balances.
+     * NOTE: make sure to call `accrueInterest()` on respective market before calling these helpers
+     */
+    function _userBorrowBalance(Id _id, address _user) internal view returns (uint256) {
+        Market memory market = morphoBlue.market(_id);
+        return (
+            uint256((morphoBlue.position(_id, _user).borrowShares)).toAssetsUp(
+                market.totalBorrowAssets,
+                market.totalBorrowShares
+            )
+        );
+    }
+
+    /**
+     * @notice Caller calls `accrueInterest` on specified MB market
+     * @param _market The specified MB market
+     */
+    function _accrueInterest(MarketParams memory _market) internal virtual {
```

Comment:
This accrueInterest is not validating market input.

Reply:
It validates it within the external `accrueInterst()` calls that then call this internal function. I looked at bringing over the `accrueInterest()` function over as a helper that is external but since it calls `validateMBMarket` for each respective adaptor, it seems that it could be simpler / work with separate external `accrueInterest()` calls within each adaptor.

Reply:
Also I do agree that putting `accrueInterest` as a helper would make sense overall (if the validate market aspect wasn't in the way). Although that would mean the `morphoBlueCollateralAdaptor` would also have the `accrueInterest()` function. This is fine, just a bit redundant, but I guess since a cellar would likely have a position with collateral in order to take out debt, then we're good. The aspect of zero LLTV markets might also bring up a different scenario, but that's TBD I guess.